src/shared/lib/oql/oqlfilter.ts

```diff
@@ -672,10 +693,44 @@ function isDatumWantedByOQLMutationModifier<T>(
     datum: T,
     accessors: IAccessorsForOqlFilter<T>
 ) {
-    switch (modifier) {
+    switch (modifier.type) {
         case 'GERMLINE':
         case 'SOMATIC':
-            return accessors.mut_status(datum) === modifier.toLowerCase();
+            return accessors.mut_status(datum) === modifier.type.toLowerCase();
+        case 'RANGE':
```

Comment:
This logic might worth to modularize in `cbioportal-utils` for future use. We can define some generic RangeUtils function. At this point it might be over-engineering though.

There might also be node packages for this. For example, I was using https://guava.dev/releases/19.0/api/docs/com/google/common/collect/Range.html for backend. But of course, that's java :)